src/index.js

```diff
@@ -102,23 +105,29 @@ async function fetchFeaturedRooms() {
   const socket = await connectToReticulum();
 
   authChannel.setSocket(socket);
-  if (authChannel.signedIn) {
-    const retPhxChannel = socket.channel(`ret`, { hub_id: "index", token: store.state.credentials.token });
-    retPhxChannel.join().receive("ok", () => {
-      retPhxChannel.push("refresh_perms_token").receive("ok", ({ perms_token }) => {
-        const perms = jwtDecode(perms_token);
-        configs.setIsAdmin(perms.postgrest_role === "ret_admin");
-
-        if (perms.postgrest_role === "ret_admin") {
-          showAdmin = true;
-          remountUI();
-        }
-
-        retPhxChannel.leave();
-      });
-    });
+  const joinParams = { hub_id: "index" };
+
+  if (store.state.credentials && store.state.credentials.token) {
+    joinParams.token = store.state.credentials.token;
   }
 
+  const retPhxChannel = socket.channel("ret", joinParams);
+  retPhxChannel.join().receive("ok", () => {
+    retPhxChannel.push("refresh_perms_token").receive("ok", ({ perms_token }) => {
+      const perms = jwtDecode(perms_token);
+      configs.setIsAdmin(perms.postgrest_role === "ret_admin");
+
+      if (perms.postgrest_role === "ret_admin") {
+        showAdmin = true;
+      }
+
+      showCreate = !!perms.create_hub;
+      remountUI();
```

Comment:
I wonder if we should wait for perms before the initial UI mount, since it may be a bit odd to have the primary CTA show up after some delay.

Reply:
Yeah I adjusted the CSS to reduce the visual effect of it appearing late, seems a tough call. A full white screen is probably bad, but maybe having the content pop in (eg background + header/footer shown) makes more sense. Also the recent changes w favorites loading may mitigate this since Dom is going to add an additional pre-load mount.